### PR TITLE
refactor:  return coordinate objects

### PR DESCRIPTION
**Pull request by nden**
This is mostly a refactoring PR but I'd love to hear comments.
The main changes are to the interface, specifically how to return coordinate objects instead of numerical values. 

Examples:

```
# return numerical values
numerical_values = wcsobj(x, y)

# return coordinates.SkyCoord objects or Quantities (for spectral data)
coords = wcsobj(x, y, output="numericals_plus")
```

The `CoordinateFrame.transform_to` method was removed as it is not convenient and duplicates astropy functionality.

There was also some code and test clean-up.

### CODE REVIEW

**Comment by nden** [comment]
@brechmos-stsci @perrygreenfield @larrybradley @stscirij 
Poking randomly people for comments on the interface change.

File: gwcs/coordinate_frames.py [hunk 1]
```diff
@@ -396,11 +296,15 @@ def __init__(self, frames, name=None):
         for frame in frames:
             axes_order.extend(frame.axes_order)
         for frame in frames:
-            for ind, type, u, n in zip(frame.axes_order, frame.axes_type,
-                                          frame.unit, frame.axes_names):
+            for ind, type, un, n in zip(frame.axes_order, frame.axes_type,
+                                        frame.unit, frame.axes_names):
```
Comment (brechmos-stsci) [inline gwcs/coordinate_frames.py]: While you are modifying things in here, you might want to change "type" here to some other variable name as "type" is a common function and could cause odd collisions.  Maybe "axtype"

Comment (nden) [inline gwcs/coordinate_frames.py]: done

File: gwcs/utils.py [hunk 2]
```diff
@@ -402,7 +391,7 @@ def fitswcs_nonlinear(header):
     sky_axes, _ = get_axes(wcs_info)
     phip, lonp = [wcs_info['CRVAL'][i] for i in sky_axes]
     # TODO: write "def compute_lonpole(projcode, l)"
-    # Set a defaul tvalue for now 
+    # Set a defaul tvalue for now
```
Comment (larrybradley) [inline gwcs/utils.py]: default value

**Pull request merged**